src/branch/context.rs

```diff
@@ -121,13 +121,12 @@ impl Context {
                 name: _name,
             } if self.project_dir.is_some() => {
                 // only place to store the branch is the database file in the project
```

Comment:
I wonder if it's worth adding to this comment to make it clear why you have this two-step process.